C++/source/sorting/bucket-sort.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+    Bucket sort
+    -----------
+    A sorting algorithm that works by distributing the elements of an array into a number of buckets.
+    Each bucket is then sorted individually, either using a different sorting algorithm, or by recursively
+    applying the bucket sorting algorithm.
+    Time complexity
+    ---------------
+    O(N^2), where N is the number of elements.
+    Space complexity
+    ----------------
+    O(N), where N is the number of elements.
+*/
+
+#include <iostream>
+#include <algorithm>
+#include <vector>
+using namespace std;
+ 
+void bucketSort(float arr[], int n)
+{
+    vector<float> b[n];
```

Comment:
It's better to use vectors (`std::vector`) instead of C-style arrays. This would also eliminate the need of another argument for the size of the array.